arches_lingo/src/arches_lingo/components/generic/LabelEditor.vue

```diff
@@ -0,0 +1,334 @@
+    ResourceInstanceResult,
+    SchemeInstance,
+} from "@/arches_lingo/types.ts";
+import type { Language } from "@/arches_vue_utils/types.ts";
+
+const emit = defineEmits(["update"]);
+const toast = useToast();
+const route = useRoute();
+const router = useRouter();
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+const { $gettext } = useGettext();
+
+const props = withDefaults(
+    defineProps<{
+        value?: AppellativeStatus;
+    }>(),
+    {
+        value: () => ({}) as AppellativeStatus,
+    },
+);
+const valueRef = toRef(props, "value");
+const formValue = computed({
+    get: () => valueRef.value,
+    set: (newVal: AppellativeStatus) => {
+        valueRef.value = newVal;
+    },
+});
+
+const languageOptions = ref<ControlledListItem[]>([]);
+const typeOptions = ref<ControlledListItem[]>([]);
+const statusOptions = ref<ControlledListItem[]>([]);
+const metatypeOptions = ref<ControlledListItem[]>([]);
+const eventTypeOptions = ref<ControlledListItem[]>([]);
+const groupAndPersonOptions = ref<ResourceInstanceReference[]>();
+const textualWorkOptions = ref<ResourceInstanceReference[]>();
+
+function onUpdateString(node: keyof AppellativeStatus, val: string) {
+    (formValue.value[node] as unknown) = toRaw(val);
+}
+
+function onUpdateReferenceDatatype(
+    node: keyof AppellativeStatus,
+    val: ControlledListItem[],
+) {
+    (formValue.value[node] as unknown) = val.map((item) => toRaw(item));
+}
+
+function onUpdateResourceInstance(
+    node: keyof AppellativeStatus,
+    val: string[],
+    options: ResourceInstanceReference[],
+) {
+    if (val.length > 0) {
+        const selectedOptions = options.filter((option) =>
+            val.includes(option.resourceId),
+        );
+        (formValue.value[node] as unknown) = selectedOptions;
+    }
+}
+
+async function save() {
+    try {
+        if (route.params.id === NEW) {
+            const newSchemeInstance: SchemeInstance = {
+                appellative_status: [toRaw(formValue.value)],
+            };
+            const updated = await createScheme(newSchemeInstance);
+            await router.push({
+                name: "scheme",
+                params: { id: updated.resourceinstanceid },
+            });
+        } else if (!formValue.value.tileid) {
+            await createSchemeLabel(route.params.id as string, formValue.value);
+        } else {
+            await updateSchemeLabel(
+                route.params.id as string,
+                formValue.value.tileid as string,
+                formValue.value,
+            );
+        }
+        emit("update");
+    } catch (error) {
+        toast.add({
+            severity: ERROR,
+            summary: $gettext("Error saving scheme"),
+            detail: (error as Error).message,
+        });
+    }
+}
+
+async function getControlledListOptions(
+    listId: string,
+): Promise<ControlledListItem[]> {
+    const parsed = await fetchControlledListOptions(listId);
```

Comment:
We don't have a global API client intercepting errors, so we have to manually show errors. Can be a follow-up.